include/gsl/span

```diff
@@ -584,7 +616,7 @@ private:
         }
```

Comment:
This function seems to be useless now: `index_type` *is* `size_t` so it needn't be narrowed to `size_t`.

Reply:
Yep, I was just about to push the removal of this.